lib/triggers/queryOrders.js

```diff
@@ -55,14 +72,50 @@ function processTrigger(msg, cfg, next, snapshot) {
             .fetch();
     }
 
+    function addReboundedOrders(ordersResponse) {
+
+        if (_.isEmpty(snapshot.reboundedOrders)) {
+            return ordersResponse;
+        }
+
+        var reboundedOrderIds = _.keys(snapshot.reboundedOrders);
+        // skip orderId-s which are in ordersResponse already
+        reboundedOrderIds = _.reject(reboundedOrderIds, function isRetreived(orderId) {
+            return _.findWhere(ordersResponse.body.results, {id: orderId});
```

Comment:
You can avoid executing ``_.findWhere`` every time by doing it once out of the loop with ``var orderResponseIds = u.pluck(ordersResponse.body.results, "id")`` and then returning in the loop ``return orderResponseIds.indexOf(orderId);``